### ISSUE TITLE
`instanceof` ActionFailure refers to the wrong object.

### ISSUE THREAD

**Issue by Curstantine** [sveltejs/kit#10361]
### Describe the bug

When you use `instanceof` on an `ActionFailure` class, it fails with a `TypeError: Right-hand side of 'instanceof' is not an object`.

### Reproduction

https://gitlab.com/Curstantine/sveltekit-instanceof-runtime

### Logs

```Shell
TypeError: Right-hand side of 'instanceof' is not an object
    at default (/home/<REDACTED>/Code/Experimental/sveltekit-instanceof-runtime/src/routes/+page.server.js:15:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Module.handle_action_json_request (/home/<REDACTED>/Code/Experimental/sveltekit-instanceof-runtime/node_modules/.pnpm/@sveltejs+kit@1.22.1_svelte@4.0.5_vite@4.4.2/node_modules/@sveltejs/kit/src/runtime/server/page/actions.js:57:16)
    at async resolve (/home/<REDACTED>/Code/Experimental/sveltekit-instanceof-runtime/node_modules/.pnpm/@sveltejs+kit@1.22.1_svelte@4.0.5_vite@4.4.2/node_modules/@sveltejs/kit/src/runtime/server/respond.js:412:18)
    at async Module.respond (/home/<REDACTED>/Code/Experimental/sveltekit-instanceof-runtime/node_modules/.pnpm/@sveltejs+kit@1.22.1_svelte@4.0.5_vite@4.4.2/node_modules/@sveltejs/kit/src/runtime/server/respond.js:279:20)
    at async file:///home/<REDACTED>/Code/Experimental/sveltekit-instanceof-runtime/node_modules/.pnpm/@sveltejs+kit@1.22.1_svelte@4.0.5_vite@4.4.2/node_modules/@sveltejs/kit/src/exports/vite/dev/index.js:505:22
```

### System Info

```Shell
System:
    OS: Linux 6.4 Arch Linux
    CPU: (4) x64 Intel(R) Core(TM) i5-3210M CPU @ 2.50GHz
    Memory: 1.95 GB / 7.65 GB
    Container: Yes
    Shell: 3.6.1 - /bin/fish
  Binaries:
    Node: 20.4.0 - /usr/bin/node
    Yarn: 1.22.19 - /usr/bin/yarn
    npm: 9.8.0 - /usr/bin/npm
    pnpm: 8.6.6 - /usr/bin/pnpm
  npmPackages:
    @sveltejs/adapter-auto: ^2.0.0 => 2.1.0 
    @sveltejs/kit: ^1.20.4 => 1.22.1 
    svelte: ^4.0.0 => 4.0.5 
    vite: ^4.3.6 => 4.4.2 
```

### Severity

annoyance

### Additional Information

This is most likely due to [this](https://github.com/sveltejs/kit/blob/9d49886f8377cf61ed632a4845912623ce9f8023/packages/kit/src/runtime/control.js#L59) "grotesque" hack not working.

This is apparent when you directly call an `ActionFailure`.
e.g.
```js
import { ActionFailure } from "@sveltejs/kit";

const x = new ActionFailure(422, { message: "Deformed request" })`; // fails here
console.debug(x instanceof ActionFailure);
```

**Comment by david-plugge** [comment]
The actual classes are not exported, just their types. This is by design and should error on the type level, no idea when this stopped doing so.

**Comment by Curstantine** [comment]
True, only the types are exported. But `instanceof` equality should work(?) according to the [comment](https://github.com/sveltejs/kit/blob/61c3e5a574d845cd4f43e53ed510155f4b854e7e/packages/kit/src/exports/vite/dev/index.js#L348) I mentioned before.

**Comment by david-plugge** [comment]
That comment is about internal stuff. The classes are loaded in nodejs and vite which means they are not the exact same class/reference.

**Comment by Curstantine** [comment]
Hmm, I see. I think it would be nice to export those classes only as types so the typescript server would complain while trying to instantiate them.

I'll leave the issue open for now.

**Comment by tomatrow** [comment]
This can be used as a temporary workaround.

```ts
import { fail, type ActionFailure as ActionFailureType } from "@sveltejs/kit"

interface Class<T> extends Function {
	new (...args: any[]): T
}

export const ActionFailure = fail(Infinity).constructor as unknown as Class<ActionFailureType>
```

**Comment by SBHattarj** [comment]
You can actually do something like this which I think looks better( I was trying to get the constructor for a library of mine )

```js
import { fail, type ActionFailure as ActionFailureType } from "@sveltejs/kit"

export const ActionFailure = fail(Infinity).constructor as typeof ActionFailureType
```

But it'd be great if we didn't had to do this honestly.

**Comment by dummdidumm** [comment]
Adding this to the 2.0 milestone - it would be good to fix the type generation or the source of the type generation such that `ActionFailure` is no longer an interface, which is wrong

**Comment by Rich-Harris** [comment]
fixed in #11261, closing
